bookkeeper-server/src/main/java/org/apache/bookkeeper/proto/BookieNettyServer.java

```diff
@@ -272,7 +363,8 @@ protected void initChannel(LocalChannel ch) throws Exception {
         }
     }
 
-    void start() {
+    void start() throws InterruptedException {
+        listenOn(bindAddress, bookieAddress);
```

Comment:
Didn't quite get it. The intention to move listenOn to start() was so that if there is an error between initialization and start() we need not bind the port and later cleanup.